### PR TITLE
#2509: Make it to return Option<EO> instead of just EO

### LINKED ISSUE #2509: Make rust foo functino return Option<EO>
In order to use `?` operator- rust feature, we should make the function return Option or Result. It will make error handling much more convenient

### PR DESCRIPTION
**Pull request by levBagryansky**
Closes #2509

---

## PR-Codex overview

> The following files were skipped due to too many changes: `eo-runtime/src/test/eo/org/eolang/rust-tests.eo`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by levBagryansky** [comment]
@Graur please check

**Comment by levBagryansky** [comment]
@Graur please check

File: eo-runtime/src/test/eo/org/eolang/rust-tests.eo [hunk 1]
```diff
@@ -256,10 +257,10 @@
     use eo_env::EOEnv;
     use eo_env::eo_enum::EO;
     use eo_env::eo_enum::EO::{EOInt};
-    pub fn foo(env: &mut EOEnv) -> EO {
-      let v = env.find("$.^.a") as u32;
-      let _bytes = env.dataize(v).unwrap();
-      EOInt(v as i64)
+    pub fn foo(env: &mut EOEnv) -> Option<EO> {
+      let v = env.find("$.^.a")?;
+      let _bytes = env.dataize(v)?;
```
Comment (Graur) [inline eo-runtime/src/test/eo/org/eolang/rust-tests.eo:262]: @levBagryansky Should there also be 4 spaces missing at the beginning of the line with the code?

File: eo-runtime/src/test/eo/org/eolang/rust-tests.eo [hunk 2]
```diff
@@ -206,15 +210,12 @@
...
-      if(env.bind(v1 , v2, "EO-att")).is_none(){
-        return EOError("bind failed".to_string());
-      } else {
-        return EOInt(0 as i64);
-      }
+    use eo_env::eo_enum::EO::{EOInt};
+    pub fn foo(env: &mut EOEnv) -> Option<EO> {
+      let v1 = env.find("$.^.a")?;
+      let v2 = env.find("$.^.b")?;
+      env.bind(v1 , v2, "EO-att")?;
```
Comment (Graur) [inline eo-runtime/src/test/eo/org/eolang/rust-tests.eo:217]: @levBagryansky Should there also be 4 spaces missing at the beginning of the line with the code?

File: eo-runtime/src/test/eo/org/eolang/rust-tests.eo [hunk 3]
```diff
@@ -307,8 +308,8 @@
     use eo_env::eo_enum::EO;
     use eo_env::eo_enum::EO::{EOError};
 
-    pub fn foo(_env: &mut EOEnv) -> EO {
-      EOError("Custom error".to_string())
+    pub fn foo(_env: &mut EOEnv) -> Option<EO> {
+      Some(EOError("Custom error".to_string()))
```
Comment (Graur) [inline eo-runtime/src/test/eo/org/eolang/rust-tests.eo:312]: @levBagryansky Should there also be 4 spaces missing at the beginning of the line with the code?

File: eo-runtime/src/test/eo/org/eolang/rust-tests.eo [hunk 4]
```diff
@@ -335,14 +336,11 @@
...
-      let copy = env.copy(eobytes as u32).unwrap();
-      if env.put(copy.clone(), &[0x00, 0x1a, 0xEE]).is_none() {
-        EOError("put failed".to_string())
-      } else {
-        EOVertex(copy)
-      }
+    pub fn foo(env: &mut EOEnv) -> Option<EO> {
+      let eobytes = env.find("Q.org.eolang.bytes")?;
+      let copy = env.copy(eobytes)?;
+      env.put(copy.clone(), &[0x00, 0x1a, 0xEE])?;
```
Comment (Graur) [inline eo-runtime/src/test/eo/org/eolang/rust-tests.eo:342]: @levBagryansky Should there also be 4 spaces missing at the beginning of the line with the code?

File: eo-runtime/src/test/eo/org/eolang/rust-tests.eo [hunk 5]
```diff
@@ -358,17 +356,14 @@
...
-        EOError("bind failed".to_string())
-      } else {
-        EOVertex(copy)
-      }
+    pub fn foo(env: &mut EOEnv) -> Option<EO> {
+      let eobook = env.find("$.^.book")?;
+      let copy = env.copy(eobook)?;
+      let eoline = env.find("$.^.line")?;
+      env.bind(copy.clone(), eoline, "content")?;
+      Some(EOVertex(copy))
```
Comment (Graur) [inline eo-runtime/src/test/eo/org/eolang/rust-tests.eo:366]: @levBagryansky Should there also be 4 spaces missing at the beginning of the line with the code?

**Review by Graur — CHANGES_REQUESTED** [review]

**Review by Graur — APPROVED** [review]
@levBagryansky Looks good to me!
@yegor256 Please have a look

**Comment by levBagryansky** [comment]
@yegor256 please take a look

**Comment by levBagryansky** [comment]
@yegor256 please check

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**